src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -84,6 +111,28 @@ private String readBytesToString() {
         return "";
     }
 
+    /**
+     * Get the hash of the logfile with target lines.
+     * @param chan a nyte channel that allow us to read the file
+     * @return the calculated hash value of the logfile, empty string if not enough lines for digest.
+     */
+    public String hashStringWithChannel(SeekableByteChannel chan) {
```

Comment:
You could just call this `hashString` and overload the other method. Ideally we delete the other method maybe on a next PR